Added DevTools landing page & bucket overview pages (#1665)
MicrosoftDocs/edge-developer: microsoft-edge/test-and-automation/test-and-automation.md

---

> title: Test and automation in Microsoft Edge

Reviewer: @bwalderman would you mind please giving this test automation intro page a quick read. It's supposed to serve as a welcome/intro/landing page for the things we have under the `Test and automation` table of content.

Author: @bwalderman - rendered page for review:
https://review.docs.microsoft.com/en-us/microsoft-edge/test-and-automation/test-and-automation?branch=pr-en-us-1665

description: DevTools Protocol instruments, debugs, profiles browsers.  Origin Trials to try experimental APIs.  Playwright cross-browser automation via 1 API.  Puppeteer API automates via DevTools Protocol.  WebDriver simulates user interaction.  webhint linting checks for errors, best practices.
author: MSEdgeTeam
ms.author: msedgedevrel
ms.topic: conceptual
ms.prod: microsoft-edge
ms.technology: devtools
ms.date: 02/14/2022
---
# Test and automation in Microsoft Edge

There are many tools to automate your testing of Microsoft Edge:

| Tool | Description |
| --- | --- |
| DevTools Protocol | Instrument, inspect, debug, and profile browsers including Microsoft Edge. |
| Origin Trials | Try out experimental APIs on live sites for a limited period of time. |
| Playwright | The Playwright library provides cross-browser automation through a single API. |

> | Puppeteer | The Puppeteer library provides a high-level API to control Microsoft Edge using the DevTools Protocol. |

Reviewer 2: ```suggestion
| Puppeteer | The Puppeteer library provides a high-level API to control browsers, including Microsoft Edge, using the DevTools Protocol. |
```

Author: Applied fix 3x: 
this summary table, 
the h2 below, and 
the Puppeteer article.

<!-- ====================================================================== -->
## Origin Trials

You can use Origin Trials to try out experimental APIs on live sites for a limited period of time.  When using Origin Trials, users of Microsoft Edge that visit your site may run code that uses experimental APIs.  To access the experimental APIs on each user machine, you don't need to go to `edge://flags` and turn on feature flags.

See [Use Origin Trials in Microsoft Edge](../origin-trials/index.md).

<!-- ====================================================================== -->
## Playwright

The Playwright library provides cross-browser automation through a single API.  Playwright enables cross-browser web automation that is evergreen, capable, reliable, and fast.

Playwright launches browsers as headless, by default.  Headless browsers don't display a UI; so generally you use the command line - however, you can also configure Playwright to run the full Microsoft Edge UI.

See [Use Playwright to automate and test in Microsoft Edge](../playwright/index.md).

<!-- ====================================================================== -->
## Puppeteer

> The Puppeteer library provides a high-level API to control Microsoft Edge using the DevTools Protocol.

Reviewer 2: ```suggestion
The Puppeteer library provides a high-level API to control Chromium-based browsers like Microsoft Edge using the DevTools Protocol.
```

Author: Applied fix 2x:
here, 
top of Puppeteer article.

Puppeteer launches headless browsers by default.  Headless browsers don't display a UI, so you must use the command line.  You can also configure Puppeteer to run full (non-headless) Microsoft Edge.

With Microsoft Edge, you can use `puppeteer-core`, a lightweight version of Puppeteer that launches an existing browser installation, like Microsoft Edge.

See [Puppeteer overview](../puppeteer/index.md).

<!-- ====================================================================== -->

> ## Use WebDriver to automate Microsoft Edge

Reviewer 2: ```suggestion
## WebDriver
```

I think this aligns better with the other headers in this file.

> WebDriver allows you to automate Microsoft Edge by simulating user interaction.  WebDriver's testing and simulation has advantages over JavaScript unit tests:

Reviewer 2: ```suggestion
WebDriver allows you to automate Microsoft Edge by simulating user interaction.  Tests that use WebDriver have some advantages over JavaScript unit tests that run in the browser:
```

Author: Applied fix 2x:
here
top of WebDriver article
